plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/generated/LocationComponentAttributeParser.kt

```diff
@@ -30,6 +31,10 @@ internal object LocationComponentAttributeParser {
         pulsingMaxRadius = typedArray.getDimension(R.styleable.mapbox_MapView_mapbox_locationComponentPulsingMaxRadius, 10f * pixelRatio),
         layerAbove = typedArray.getString(R.styleable.mapbox_MapView_mapbox_locationComponentLayerAbove),
         layerBelow = typedArray.getString(R.styleable.mapbox_MapView_mapbox_locationComponentLayerBelow),
+        puckBearingSource = when (typedArray.getInt(R.styleable.mapbox_MapView_mapbox_locationComponentPuckBearingSource, 0)) {
+          0 -> PuckBearingSource.HEADING
+          else -> PuckBearingSource.COURSE
```

Comment:
Looking at generated code for `LocationPuck` below - we do init with `-1`, have 2 branches covered:
0 -> 2D puck
1 -> 3D puck
else -> 2D puck
why don't we have similar structure here?

Reply:
I have noticed this while writing this part, but IMO this way is more clear and simple. But we still need to update this part and put `PuckBearingSource.HEADING` in the else block for the default value.